02.calendar/calendar.rb

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+
+opt = OptionParser.new
+option = {}
+
+opt.on('-y [v]') {|v| option[:y] = v}
+opt.on('-m [v]') {|v| option[:m] = v}
+
+opt.parse!(ARGV)
+
+year_num = option[:y].to_i
+month_num = option[:m].to_i
+
+today = Date.today
+
+if year_num > 0 && month_num > 0
+  @run_today = Date.new(year_num, month_num)
+elsif year_num > 0
+  @run_today = Date.new(year = year_num, mon = today.month)
+elsif month_num > 0
+  @run_today = Date.new(year = today.year, mon = month_num)
```

Comment:
引数の書き方がちょっとおかしいですね。これでも一応動くといえば動きますが、引数の中で`=`による代入を行うことは基本的にないです。つまりこうであるべきです。

```suggestion
  @run_today = Date.new(year_num, today.month)
elsif month_num > 0
  @run_today = Date.new(today.year, month_num)
```

なぜこの書き方に至ったのか、過程が少し気になっています。もしかすると、[公式リファレンス](https://docs.ruby-lang.org/ja/latest/class/Date.html#S_CIVIL)にある`new(year = -4712, mon = 1, mday = 1, start = Date::ITALY)`のような記述を参考にしたということでしょうか？

もしそうであれば、これは「引数のデフォルト値」というやつですので代入とは違います。ここら辺がもし曖昧になっているようでしたらもう一度調べてみてください〜

Reply:
どうしても見てわかりやすいような記載をしてしまい基本的な記法より逸脱してしまった！

Reply:
## ログ
修正：済